ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/common/GenesisWorldStateProvider.java

```diff
@@ -57,7 +61,7 @@ public static MutableWorldState createGenesisWorldState(
    *
    * @return a mutable world state for the Genesis block
    */
-  private static MutableWorldState createGenesisBonsaiWorldState() {
+  private static MutableWorldState createGenesisBonsaiWorldState(final boolean archiveMode) {
```

Comment:
nit : why not passing directly DataStorageConfiguration instead of the boolean ?